nano/node/transport/tcp.cpp

```diff
@@ -160,6 +178,26 @@ void nano::transport::tcp_channels::stop ()
 	channels.clear ();
 }
 
+// TODO: Merge with keepalive in network class
+void nano::transport::tcp_channels::run_keepalive ()
+{
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	while (!stopped)
+	{
+		condition.wait_for (lock, node.network_params.network.keepalive_period);
```

Comment:
It probably doesn't make much of a difference but wait_for can spurriously wake up.

Reply:
It's a good point. Here I wanted something that works and is simple to implement. It should be trivial to implement a helper that ensures the interval is always respected. Adding this to my todo list.